jsonargparse_tests/test_dataclass_like.py

```diff
@@ -978,3 +984,13 @@ def test_field_init_false(self, parser):
         assert cfg == Namespace()
         init = parser.instantiate_classes(cfg)
         assert init.data.p1 == {}
+
+    def test_nested_with_default(self, parser):
+        parser.add_argument("--data", type=AttrsWithNestedDefaultDataclass)
+        cfg = parser.parse_args(["--data.p1=1.23"])
+        assert cfg.data == Namespace(p1=1.23, subfield=Namespace(p1="-", p2=0))
+
+    def test_nested_without_default(self, parser):
+        parser.add_argument("--data", type=AttrsWithNestedDataclassNoDefault)
+        cfg = parser.parse_args(["--data.p1=1.23"])
+        assert cfg.data == Namespace(p1=1.23, subfield=Namespace(p1="-", p2=0))
```

Comment:
This looks somewhat unexpected. I would have guessed that the parsed value would be `Namespace(p1=1.23, subfield=None)`. But anyway, this is not related to this pull request so I would say to keep it like this for now. Probably this behavior will need to change for #287.